File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <unordered_set>

using namespace std;

string solution(string s, string skip, int index) {
    string answer = "";
    
    unordered_set<char> skip_set(skip.begin(), skip.end());
    
    for(auto& c : s){
        
        char cur = c;
        int cnt = 0;
        
        while(cnt < index){
            cur++;
            if(cur > 'z')
                cur = 'a';
            if(skip_set.find(cur) == skip_set.end())
                cnt++;
        }
        
        answer.push_back(cur);
    }
    
    return answer;
}

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public String solution(String s, String skip, int index) {
        
        List<Character> list = new ArrayList<>();
        Set<Character> set = skip.chars().mapToObj(ch -> (char)ch)
            .collect(Collectors.toSet());
        
        for(int i = 0 ; i < 26 ; i++){
            char cur = (char)('a' + i);
            if(!set.contains(cur))
                list.add(cur);
        }
        
        return s.chars().mapToObj(ch -> (char)ch)
            .map(ch -> list.get((list.indexOf(ch) + index) % list.size()))
            .map(String::valueOf)
            .collect(Collectors.joining());
    }
}